This notebook is an exploration of the structure of the CORD-19 data files.

# Imports

In [1]:
import csv
import pandas as pd
from pprint import pprint as pp

# metadata.csv

In [2]:
metadataFile = open('metadata.csv', 'r')
csvReader = csv.reader(metadataFile, delimiter=',', quotechar='"')

Header = next(csvReader)
pp(Header)

['cord_uid',
 'sha',
 'source_x',
 'title',
 'doi',
 'pmcid',
 'pubmed_id',
 'license',
 'abstract',
 'publish_time',
 'authors',
 'journal',
 'mag_id',
 'who_covidence_id',
 'arxiv_id',
 'pdf_json_files',
 'pmc_json_files',
 'url']


## Columns

Starting from the readme provided with the data (metadata.readme) we can outline what each columns should cover and, importantly for databasing it, whether it is single value of potentially multi-valued.

'cord_uid' - This is a persistent identifier for the article within the CORD-19 ecosystem. __single valued__, __unique__.

'sha' - The hash of the article's PDF. There may be multiple PDFs associated to one article (supporting materials, ?preprints?). __multi-valued__.

'source_x' - The source of the article data. Sources covered include CZI (Chan-Zuckerberg Initiative), PMC (pubmed central), bioRxiv and medRxiv. Confirm sources from data. Should be __single valued__.

'title' - The article's title. __single valued__.

'doi' - The article's doi. __single valued__.

'pmcid' - The article's pubmed central id, if it has one. __single valued__, __unique__.


'pubmed_id' - The article's pubmed id. For distinction between this and the previous, see [here](https://publicaccess.nih.gov/include-pmcid-citations.htm#Difference). __single valued__, __unique__.

'license' - The license under which the data is being shared. Should be __single valued__.

'abstract' - The article's abstract. __single valued__.

'publish_time' - The date the article was published. __single valued__.

'authors' - The article's authors. Coverage questionable. __multi-valued__.

'journal' - The journal in which the article was published. If it was published in a journal. What happens with preprints? __single valued__.

'mag_id' - The article's microsoft academic graph id. __single valued__, __unique__.

'who_covidence_id' - The article's WHO #Covidence id. Should only be populated for CZI source articles. __single valued__, __unique__.

'arxiv_id' - The article's arXiv id. __single valued__, __unique__.

'pdf_json_files' - Relative path of the json(s) parsed from pdf file(s). __multi-valued__, __unique__.

'pmc_json_files' - Relative path of the json(s) parsed from pmc file(s). __multi-valued__, __unique__.

'url' - The article's URL. Should be __single valued__, __unique__.

To explore the columns, it is probably easier to work with the csv via pandas.

In [3]:
metadataFrame = pd.read_csv('metadata.csv', low_memory=False)
metadataFrame

cord_uid                                                sha  \
0      zjufx4fo           b2897e1277f56641193a6db73825f707eed3e4c9   
1      ymceytj3           e3d0d482ebd9a8ba81c254cc433f314142e72174   
2      wzj2glte           00b1d99e70f779eb4ede50059db469c65e8c1469   
3      2sfqsfm1           cf584e00f637cbd8f1bb35f3f09f5ed07b71aeb0   
4      i0zym7iq           dde02f11923815e6a16a31dd6298c46b109c5dfa   
5      xqhn0vbp           1e1286db212100993d03cc22374b624f7caee956   
6      gi6uaa83           8ae137c8da1607b3a8e4c946c07ca8bda67f88ac   
7      0m32ecnu           23bc55d6f63fab18b02004483888db2b6a0bfa48   
8      le0ogx1s                                                NaN   
9      3oxzzxnd           7ff45096210eeb392d51f646f5c7fe011079aaf3   
10     fy4w7xz8           0104f6ceccf92ae8567a0102f89cbb976969a774   
11     0qaoam29           5b68a553a7cbbea13472721cd1ad617d42b40c26   
12     qj4dh6rg           3ed670f60a7be2e3e2a991ea8af1fdd5fa5e2b2c   
13     1wswi7us           59ffca26c9d1f881e4e37e7c5af626d72d858dc4   
14     yy96yeu9           c63c4d58d170136b8d3b5a66424b5ac3f73a92d9   
15     5o38ihe0           dab2d95da5307c840eb4a1e00f0186022541d63d   
16     1ul8owic           75ebad1e8641e5fe2a51214f4d97742ab59a8dae   
17     5s6acr7m           6c0dee643345c93436cfa8d58270aa887ad92e3d   
18     6lvn10f4           14e0cac6e86d62859e6c9f1351ab67466d89e5b3   
19     tvxpckxo           d09b79026117ec9faebba46a8d13aa9b23ec751e   
20     mcuixluu           44102e3e69e70ad2a73e753133283334ce1f8736   
21     ri5v6u4x           06559dd625491d6474ee88f08c12fc17c1830995   
22     kuybfc1y           d81ae79ebc6462fd3de075e220992f1aae1f06ae   
23     ng4rrdte           87390d2ae28407b3e03e60a6b24a7fd99ed7229a   
24     1769ovyk           c5131e5f5c6000ec84139edc64778a6f1d391b83   
25     tixxm78q           8a7d5de5ea680e784ab2bd877240bf09e4c1c02d   
26     pwsvhitd           e4b48ce0579f908da6dd3289bb2cc262dbdeae72   
27     wt8zfqk0           52ce19fe79de46d9bb0fa7e75a0631fed2d3efd2   
28     5gsbtfag           b499a41ad1d85a0e071b59e60065c874e534911d   
29     fpj5urao  4a228865af2c19adf9386a5b04cca3ceb3c8683d; e9c7...   
...         ...                                                ...   
63541  othckgtm                                                NaN   
63542  pth2d40p  29d15895b6f3c6472f3bab820d6a99feefae60df; 1f5c...   
63543  qko8md3w           a68721d234eacc8dd31ff6a97616d8091f61e912   
63544  aq7amykr           1e23bde4e435cbcb5f23f6b1dc2f6687f1c3b221   
63545  5wrj838n           66a268ec094c120ea3c4dfeedfeacab4494c50f6   
63546  ccdj7137           8d1498b4b95b0bf8f6d9577764fffcecd1bae360   
63547  tjy7f9rk           258ce44a78453f7d28e241962a197995d3e53c04   
63548  56ehhgmk           9074747aea8d86b85b9594e82ccc2fa44d08772e   
63549  drxpor44           ba1ab411e62b278e72259c8203b76991c3efbc7d   
63550  pcnmbd72           3d678ce5b0b21cca48a07226a97055eb96733352   
63551  jwlwrz8v           b1a4463a80b51e113e68c992b53781869881b73a   
63552  xl5z5lsk           9fd57e68de682f9cf91b1ec2479868fd7215c0ed   
63553  0h7s5d7n  e44284f191fc9522ccc675257354cb33cd9c2885; 26de...   
63554  zfe4lbr0           2e7f047e863ea96cf05fe606a7e8163d3217e1ab   
63555  iegx2yln                                                NaN   
63556  vyu0zd6k  a243da5abdf029948b1fa8beb600db0ab124da63; 7df3...   
63557  2ho8sr1a           613c002da74850a9055fceaabb8a20d27c2c69b7   
63558  d7n1zy0b           fbd0883af84e01bf3f6c874e9cb863934995a113   
63559  sllxevf7           d66311a0656685a7a39bd82b24ac3cb16c17ecfa   
63560  zi8ij7ln                                                NaN   
63561  54gh3rt4           d0f619ec08f0d63dd120c0acf095493a0d869935   
63562  m45vm6sf                                                NaN   
63563  sehhfysa  78c0852bc3566669cb9f7903758a92988362bcf9; 64f2...   
63564  01plpnzu           f5110e5b6c82b1000b924e550afd1722dac5cda8   
63565  89t3g9ty           1c2ec5f7aa7133b58659aa5403e03f14da7ba1a1   
63566  mvq6353a  

First start by making sure the columns that are supposed to be single valued and unique are actually so.

In [4]:
SingleValued_and_Unique = ['cord_uid', 'pmcid', 'pubmed_id', 'mag_id', 'who_covidence_id', 'arxiv_id', 'url']

In [5]:
for Column in SingleValued_and_Unique:
    print(Column, metadataFrame[Column].dropna().is_unique)

cord_uid False
pmcid True
pubmed_id False
mag_id False
who_covidence_id True
arxiv_id True
url False


All except who_covidence_id and arxiv_id appear to be non-unique. Isn't that just wonderful...

Start by checking the cord_uid.

In [6]:
ValueCounts = metadataFrame['cord_uid'].value_counts()
ValueCounts[ValueCounts>1]

fzq71ghi    8
jsk1oztb    3
0klupmep    2
hox2xwjg    2
c4u0gxp5    2
j7swau26    2
adygntbe    2
5ei7iwu0    2
79mzwv1c    2
4fbr8fx8    2
o4r34pff    2
vp5358rr    2
qhftb6d7    2
m6q8kbjg    2
vqbreyna    2
6z5f2gz3    2
21htepa1    2
huablvd1    2
8fwa2c24    2
laq5ze8o    2
3ury4hnv    2
7y8fd521    2
sv9mdgek    2
xjpev4jw    2
e9pwguwm    2
mmls866r    2
0z5wacxs    2
21qu87oh    2
eich19nx    2
j3b964oz    2
6hdoap81    2
2maferew    2
4hlvrfeh    2
5kzx5hgg    2
brz1fn2h    2
30duqivi    2
940au47y    2
Name: cord_uid, dtype: int64

Ugh. As this supposed to the persistent identifier, looks like some duplicates happening.

So the easiest way to start with this is by checking to see if these are duplicates, and if they are just eliminating the double records.

In [7]:
metadataFrame[metadataFrame['cord_uid'].duplicated(keep=False)].sort_values(by=['cord_uid'])

cord_uid                                       sha  source_x  \
30977  0klupmep                                       NaN  Elsevier   
16454  0klupmep                                       NaN       PMC   
30885  0z5wacxs  7e787fd2ae5b544add6281d3d40ad322de26aa17  Elsevier   
16452  0z5wacxs  7e787fd2ae5b544add6281d3d40ad322de26aa17       PMC   
30416  21htepa1  a25e212b03cc65c44dcc336775b101934e30f041  Elsevier   
16473  21htepa1                                       NaN       PMC   
30156  21qu87oh  68249d769e1926678af8d52d2484e36787e13525  Elsevier   
42065  21qu87oh                                       NaN       PMC   
16467  2maferew                                       NaN       PMC   
30766  2maferew  6744bc52b1b29d2ab28cbaeef8942eebece5175b  Elsevier   
30994  30duqivi  4e8306655e7f3c6ae88db9708a400ab78c011700  Elsevier   
42071  30duqivi                                       NaN       PMC   
30159  3ury4hnv  5d2ed78357c0c88b4500d644f4a717d0192a0d2d  Elsevier   
42072  3ury4hnv                                       NaN       PMC   
30920  4fbr8fx8  bd3945ec343cc4535650c9425aa143f71716de7c  Elsevier   
15525  4fbr8fx8                                       NaN       PMC   
30590  4hlvrfeh  e666b6507719561e391f4246a140695b213c8e5b  Elsevier   
16470  4hlvrfeh                                       NaN       PMC   
42067  5ei7iwu0                                       NaN       PMC   
30157  5ei7iwu0  bad208dd9e4c71121ae4160ccf916c9f69eb62a3  Elsevier   
16472  5kzx5hgg                                       NaN       PMC   
30417  5kzx5hgg                                       NaN  Elsevier   
30155  6hdoap81  88a8c34283c82d0f854254d077953980954bd366  Elsevier   
42066  6hdoap81                                       NaN       PMC   
30146  6z5f2gz3  7fa6a89af06e588d737c868cdcebceaee14d7532  Elsevier   
42052  6z5f2gz3                                       NaN       PMC   
42070  79mzwv1c                                       NaN       PMC   
30145  79mzwv1c  f3d760db99cfc4224b989b1e53bec59e77823e05  Elsevier   
31020  7y8fd521  58b578c2c6619df320c9a0bcb4d434edc2300d00  Elsevier   
6994   7y8fd521                                       NaN       PMC   
...         ...                                       ...       ...   
41846  fzq71ghi                                       NaN       WHO   
42365  hox2xwjg                                       NaN       PMC   
30917  hox2xwjg  a8dbc77e172c01c4f5c548f2764286b75e618ef5  Elsevier   
30147  huablvd1  4656f1c7a371893b0b6dcb31007fe89b64ab325f  Elsevier   
42054  huablvd1                                       NaN       PMC   
17267  j3b964oz  c1402ba933f4a2a122d5ea706b19fc536568ca53       PMC   
19619  j3b964oz  346855c8950ef288f1a27e3e1a7281c7e49027c5       PMC   
30153  j7swau26  01213acdd86020357259f2a1094bc43f9bb79796  Elsevier   
42064  j7swau26                                       NaN       PMC   
28796  jsk1oztb                                       NaN       PMC   
28806  jsk1oztb                                       NaN       PMC   
28744  jsk1oztb  24cc817d27e43b9e7bdc5537872088710ea2a566       PMC   
43261  laq5ze8o                                       NaN       PMC   
45083  laq5ze8o                                       NaN       WHO   
15968  m6q8kbjg  ea2c56be23c5e3fb8261d9612585433aa12f1cec       PMC   
45076  m6q8kbjg                                       NaN       WHO   
16451  mmls866r                                       NaN       PMC   
30919  mmls866r  151b80cfa218d8a252c1720d3e2f2e7b1effd96c  Elsevier   
42289  o4r34pff                                       NaN       PMC   
30781  o4r34pff  9bc29db5852a315d576437636d2398ebf69ac52c  Elsevier   
42073  qhftb6d7                                       NaN       PMC   
30993  qhftb6d7  db732b73268d958bf73ae634a13e58df7b698c8f  Elsevier   
30976  sv9mdgek                                       NaN  Elsevier   
16453  sv9mdgek                                       NaN       PMC   
30447  vp5358rr  f3eefad19d628c

For the number there are here, it would probably be possible to merge manually or by a rule based approach.

I tried this in a previous version, but after messing with it for 15 minutes realized it is likely better to just drop these records and keep moving forward. Fewer than 50 records won't make a difference in the analysis that will be later carried out.

In [8]:
print('Rows before: {}'.format(metadataFrame.shape[0]))
metadataFrame = metadataFrame.drop_duplicates('cord_uid', keep=False) # drop all duplicates. keep='first' or keep='last' would keep the first/last
print('Rows after: {}'.format(metadataFrame.shape[0]))

Rows before: 63571
Rows after: 63490


In [9]:
metadataFrame['cord_uid'].dropna().is_unique

True

So that worked.

In [10]:
for Column in SingleValued_and_Unique:
    print(Column, metadataFrame[Column].dropna().is_unique)

cord_uid True
pmcid True
pubmed_id False
mag_id False
who_covidence_id True
arxiv_id True
url False


Also cleared up pmcid. Would have been nice if it cleared up a few of the others too though. Let's do a quick check of what discrepancies are left.

In [11]:
for Column in SingleValued_and_Unique:
    ValueCounts = metadataFrame[Column].value_counts()
    print(ValueCounts[ValueCounts>1])

Series([], Name: cord_uid, dtype: int64)
Series([], Name: pmcid, dtype: int64)
32117569.0    2
25957460.0    2
31903811.0    2
12932399.0    2
27381971.0    2
32305024.0    2
16194517.0    2
26766408.0    2
14550714.0    2
Name: pubmed_id, dtype: int64
3.006646e+09    14
3.004791e+09    11
3.006304e+09    10
3.005943e+09     9
2.604381e+09     8
3.006643e+09     8
3.005657e+09     7
3.001119e+09     7
3.004897e+09     6
3.005080e+09     6
3.005847e+09     6
3.006356e+09     6
3.006338e+09     6
3.005478e+09     5
3.005539e+09     5
3.005680e+09     5
3.004824e+09     5
3.005689e+09     4
3.006008e+09     4
3.002539e+09     4
3.004398e+09     4
3.004450e+09     4
3.003886e+09     4
3.006114e+09     3
3.006448e+09     3
2.002765e+09     3
3.005490e+09     3
3.005929e+09     3
3.005656e+09     3
3.004511e+09     3
                ..
3.006078e+09     2
3.006111e+09     2
3.006116e+09     2
3.006128e+09     2
3.006178e+09     2
3.005811e+09     2
3.006211e+09     2
3.005803e+09     2
3.0057

pubmed_id has only 9 duplicates. Am inclined to just keep these as is, but have a look.

In [12]:
metadataFrame[metadataFrame['pubmed_id'].duplicated(keep=False) & metadataFrame['pubmed_id'].notnull()].sort_values(by=['pubmed_id'])

cord_uid                                                sha  \
47312  xztmxztr                                                NaN   
51437  4cs3h4xv                                                NaN   
54607  8kw4y1f2                                                NaN   
57434  jy3t5z8g                                                NaN   
53629  sjnb5q3z           a8d38ca8527d7796d5d31a6e1f9317cb724aed26   
61122  jl4tnzg6           95f6fbabdb2b2fb7d014935e18739101752dc528   
61028  jdjdeeh1           2123a7bb916a6200697beff6b7fe1f57f1680406   
52897  q0aasznp           28f309f78ae68a7ad40bf2fb1b4cebece70c36e1   
47800  1nl7q6cy           a5399231d85b304d316778dbe87a845d81db311a   
59846  npy4cdk9           fd5cb975c746ac1f4b98d9f4b3b03eaabb997dc7   
51401  wwpywvcq           7169f856c6b021789a899bc474e3dbd4990f8ef3   
61843  0mabnu9r           8f361a795b8676db7a88cdd2fd6d30042ebf6d23   
15104  i8qvat3o           179e23cae17b70aab44cef424080cf37832c3e3f   
19427  1tn3a4qs           3431417c7db9dfad0645e3fedcf8d8857845d9d4   
31214  4bdusuqq                                                NaN   
55340  2jq626ye  5a17ed3e4abf295f5820c65f56398266c1baae98; 4890...   
60869  zhw8vh3e           36e2047d1674c3095617f3eb97f9f61e48989dfe   
62107  6donibpz           634c6bb375bb87779fac5db6ac5ad97ff65619ff   

            source_x                                              title  \
47312  Elsevier; PMC  Questions about comparative genomics of SARS c...   
51437  Elsevier; PMC  Questions about comparative genomics of SARS c...   
54607  Elsevier; PMC           Treatment of SARS with human interferons   
57434  Elsevier; PMC           Treatment of SARS with human interferons   
53629  Elsevier; PMC  Neumomediastino espontáneo: estudio descriptiv...   
61122  Elsevier; PMC  Spontaneous Pneumomediastinum: Descriptive Stu...   
61028  Elsevier; PMC  Review of Non-bacterial Infections in Respirat...   
52897  Elsevier; PMC  Revisión sobre las infecciones no bacterianas ...   
47800  Elsevier; PMC               Pediatric Asthma and Viral Infection   
59846  Elsevier; PMC                            Asma y virus en el niño   
51401  Elsevier; PMC  ¿Qué ha cambiado en la neumonía adquirida en l...   
61843  Elsevier; PMC  What has Changed in Community-Acquired Pneumon...   
15104            PMC  Progress in the development of virus-like part...   
19427            PMC  Gold nanoparticles for preparation of antibodi...   
31214            WHO  Therapeutic strategies in an outbreak scenario...   
55340       CZI; PMC  Therapeutic strategies in an outbreak scenario...   
60869  Elsevier; PMC  Artificial Intelligence (AI) applications for ...   
62107  Elsevier; PMC  Industry 4.0 technologies and their applicatio...   

                                  doi       pmcid   pubmed_id    license  \
47312   10.1016/s0140-6736(03)14130-x  PMC7134733  12932399.0      no-cc   
51437   10.1016/s0140-6736(03)14131-1  PMC7135576  12932399.0  els-covid   
54607   10.1016/s0140-6736(03)14482-0  PMC7134736  14550714.0      no-cc   
57434   10.1016/s0140-6736(03)14483-2  PMC7134646  14550714.0      no-cc   
53629                10.1157/13078656  PMC7131659  16194517.0  els-covid   
61122   10.1016/s1579-2129(06)60274-7  PMC7129626  16194517.0  els-covid   
61028      10.1016/j.arbr.2015.09.015  PMC7105177  25957460.0  els-covid   
52897    10.1016/j.arbres.2015.02.015  PMC7130696  25957460.0  els-covid   
47800      10.1016/j.arbr.2016.03.010  PMC7105201  26766408.0  els-covid   
59846    10.1016/j.arbres.2015.11.008  PMC7131251  26766408.0  els-covid   
51401    10.1016/j.arbres.2016.06.001  PMC7130473  27381971.0  els-covid   
61843      10.1016/j.arbr.2016.11.015  PMC7105175  27381971.0  els-covid   
15104   10.1080/14760584.2020.1711053  PMC7103727  31903811.0      no-cc   
19427   10.1080/14760584.2020.1758070  PMC7196924  31903811.0      no-cc   
31214  10.12688/f1000research.22211.1         NaN  32117569.0    gold-oa   
55340  10.12688/f1000research.2221

It looks like some of these are merged sources (Elsevier; PMC). I don't like how there are discrepancies on the authors in some cases though.

For now am making decision to just drop.

In [13]:
# This requires a bit more care then when we dealt above with cord_uid because here NaN is a valid value.
# But drop_duplicates counts all NaN as equal.
print('Rows before: {}'.format(metadataFrame.shape[0]))
metadataFrame = metadataFrame[~(metadataFrame['pubmed_id'].duplicated(keep=False) & metadataFrame['pubmed_id'].notnull())]
print('Rows after: {}'.format(metadataFrame.shape[0]))

Rows before: 63490
Rows after: 63472


Turning to mag_id.

In [14]:
metadataFrame[metadataFrame['mag_id'].duplicated(keep=False) & metadataFrame['mag_id'].notnull()].sort_values(by=['mag_id'])

cord_uid                                                sha  \
50637  5mo330s2                                                NaN   
31102  qzbqxjdi           c630ebcdf30652f0422c3ec12a00b50241dc9bd9   
31118  0gier0lu                                                NaN   
31202  497z31h6                                                NaN   
31237  97x8q2j9                                                NaN   
41831  0tetqt33                                                NaN   
41838  w11wop27                                                NaN   
31246  wk4zxmz9                                                NaN   
31245  sonxopa0                                                NaN   
31162  lu0tni6a                                                NaN   
31349  211qdnxv                                                NaN   
61513  oe7adj91           6ee81707d6ecfc383720d49e2c884cd28378ae02   
41804  lesa5u2n                                                NaN   
42030  8ruux4sw                                                NaN   
41936  xk5igdwo                                                NaN   
52842  zndtddty           d682f1a564a088d3d01a66f88b5c88c26b1816fd   
31203  uusi3nph                                                NaN   
41762  1hwcp09t                                                NaN   
50122  zq2h0avw  247161ad51b7f362e9cadfcd39e94dd6a1a26ca3; dbaf...   
31230  2uu5090e                                                NaN   
58036  szsb1oan           cd4f5118038ae33283d34a15f8c3888f00e247bc   
54340  0nhgxoim           ea646b114ea01f978d205ed97405e1f2afed2dfd   
31367  iprkszdn                                                NaN   
61934  ghi0ekm5           ffad7416effd57e66c6f2143ad660768af44cee9   
41790  8y3t0r2d                                                NaN   
41732  61snvddh           a3e8cfc509af101c522d4193c8917d6e8f68b37b   
55641  5kkr6bcl                                                NaN   
51988  b5fdwhl6           29798795671ad2de1774d2c389e8b7159315a9ba   
47676  a5x8bmfr           2d7449fa45ea8dd5c296f99ed1e4820dccfecc38   
61120  q92unteg           80ced0d7d0c9706800891b44a5ddcf3ce85ed40a   
...         ...                                                ...   
41929  kex1pe6f                                                NaN   
41926  uxeb72ud                                                NaN   
41992  3chs390a                                                NaN   
47946  t8nnpthm                                                NaN   
56002  n11fcwmw  6d16e166279a7d116a78a21d4bd4c00b505a617e; f758...   
58454  l5ogbl5p           2f8a485f1acf987c6fe46621216f436b0e351af6   
31296  35roxf0h                                                NaN   
31200  w2ci1bfr                                                NaN   
31275  is2aundt                                                NaN   
31234  h6jt1fj1                                                NaN   
55895  1mchg5yg           43c92bb293ea5ea1be4c5a6da11fe517b2b5677e   
41722  rkfmgz8w                                                NaN   
31096  wsfhlf47                                                NaN   
49162  3hgy4msc  938eca42d73a6e5445fc7617b37e1aae84f712a2; 46bf...   
62494  36g4zdqa           469ed0f00c09e2637351c9735c306f27acf3aace   
31298  k8dpma9y                                                NaN   
53230  caturc6q           69d308d9661cf893a6a425beb328ddafab4dae9c   
62401  64myf93w           2df23f07a12fb7aa5aba327473af415e8e40f54a   
31235  d1nph2tl                                                NaN   
51112  9hxihlw9           80734e9029dd6c1a74bf7b176d8818b199be7e4c   
53436  xvnv17zy           b11b26f30a25036589753cd3ec28b64bc2633838   
61148  87d7gzgb           6da968668b2379ce81436dcede1332563e54d334   
56615  2xhhee54           074ca9f95ff2cb97eefbfa3486ae76d460184171   
59514  z37athi5           f8256d5bd56000a15a93965527e29a182d88d377   
60280  xuczplaf           1d162cd83fb09ad4919ab8445cfdef7c6206fe50   
56965  0xhho1sh  

mag_id is just wrong a lot of the time. Will leave these records as is. I don't plan on using mag_id in anycase, so these errors shouldn't matter.

Now looking at url.

In [15]:
metadataFrame[metadataFrame['url'].duplicated(keep=False) & metadataFrame['url'].notnull()].sort_values(by=['url'])

cord_uid                                                sha  \
22899  dprfxamv           a1ffb2fbd3396e59c4cd1f85e6ac3e127b50862e   
23252  bzb7cgrq                                                NaN   
23161  pf3xdouw                                                NaN   
52285  8p9jw3gl  1aa2cc6f509964807d510b48a63180e09ee083de; 6eae...   
21418  2d6nptjf                                                NaN   
57643  gttpnxvv  8ea6e8c5dc57d3f65014a7f2a835279ae8280d71; adb3...   
20659  imne9d3m                                                NaN   
21946  hwnet5jv                                                NaN   
29755  0gamepww                                                NaN   
29764  vknvwn57                                                NaN   

            source_x                                              title  \
22899            PMC  Effectiveness of Border Screening for Detectin...   
23252            PMC  Effectiveness of Border Screening for Detectin...   
23161            PMC  SHEA/APIC Guideline: Infection Prevention and ...   
52285  Elsevier; PMC  SHEA/APIC Guideline: Infection Prevention and ...   
21418            PMC   Infection Prevention in the Emergency Department   
57643  Elsevier; PMC  Infection Prevention for The Emergency Departm...   
20659            PMC  Community-Acquired Pneumonia Requiring Hospita...   
21946            PMC  Community-Acquired Pneumonia Requiring Hospita...   
29755            PMC  Sars-CoV-2 (COVID-19) Outbreak and Breast Canc...   
29764            PMC  Lung Changes on Chest CT During 2019 Novel Cor...   

                                     doi       pmcid   pubmed_id    license  \
22899          10.2105/ajph.2012.300761r  PMC4561613  26313050.0   green-oa   
23252           10.2105/ajph.2012.300761  PMC4007855  23237174.0   green-oa   
23161                     10.1086/592416  PMC3319407  18767983.0   green-oa   
52285         10.1016/j.ajic.2008.06.001  PMC3375028  18786461.0   green-oa   
21418  10.1016/j.annemergmed.2014.02.024  PMC4143473  24721718.0   green-oa   
57643          10.1016/j.emc.2018.06.013  PMC6203442  30297010.0   green-oa   
20659              10.1056/nejmoa1500245  PMC4728150  26172429.0  bronze-oa   
21946              10.1056/nejmoa1405870  PMC4697461  25714161.0  bronze-oa   
29755           10.5152/ejbh.2020.300320  PMC7138359         NaN  bronze-oa   
29764           10.5152/ejbh.2020.010420  PMC7138361         NaN  bronze-oa   

                                                abstract publish_time  \
22899  Objectives. We measured symptom and influenza ...   2015-10-01   
23252  Objectives. We measured symptom and influenza ...   2013-08-01   
23161                                                NaN   2008-09-01   
52285                                                NaN   2008-09-01   
21418  Infection prevention remains a major challenge...   2014-09-01   
57643                                                NaN   2018-11-01   
20659  BACKGROUND: Community-acquired pneumonia is a ...   2015-07-30   
21946  BACKGROUND: U.S. incidence estimates of pediat...   2015-02-26   
29755                                                NaN   2020-04-06   
29764                                                NaN   2020-04-06   

                                                 authors  \
22899  Priest, Patricia C.; Jennings, Lance C.; Dunca...   
23252  Priest, Patricia C.; Jennings, Lance C.; Dunca...   
23161  Smith, Philip W.; Bennett, Gail; Bradley, Suza...   
52285  Smith, Philip W.; Bennett, Gail; Bradley, Suza...   
21418  Liang, Stephen Y.; Theodoro, Daniel L.; Schuur...   
57643  Liang, Stephen Y.; Riethman, Madison; Fox, Jos...   
20659  Jain, S.; Self, W.H.; Wunderink, R.G.; Fakhran...   
21946  Jain, Seema; Williams, Derek J.; Arnold, Sandr...   
29755           Çakmak, Güldeniz Karadeniz; Özmen, Vahit   
29764  Çinkooğlu, Akın; Bayraktaroğlu, Selen; Savaş, ...   

                                           journal  mag_id who_covidence_id

I'm inclined to just drop these.

In [16]:
print('Rows before: {}'.format(metadataFrame.shape[0]))
metadataFrame = metadataFrame[~(metadataFrame['url'].duplicated(keep=False) & metadataFrame['url'].notnull())]
print('Rows after: {}'.format(metadataFrame.shape[0]))

Rows before: 63472
Rows after: 63462


Have arrived at a more/less clean data set of 63462 metadata records.